libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -217,8 +218,10 @@ private List<String> mapCisToExitCodes(ContraIndicators contraIndicators)
     }
 
     private boolean breachingCiThreshold(ContraIndicators contraIndicators) {
-        return contraIndicators.getContraIndicatorScore(configService.getContraIndicatorConfigMap())
-                >= Integer.parseInt(configService.getSsmParameter(CI_SCORING_THRESHOLD));
+        return ciScoreBreachesThreshold(
+                contraIndicators.getContraIndicatorScore(
+                        configService.getContraIndicatorConfigMap()),
+                configService.getSsmParameter(CI_SCORING_THRESHOLD));
```

Comment:
It feels a bit weird to pull out just the comparison here - you've got two suspiciously similar methods: `breachingCiThreshold` and `ciScoreBreachesThreshold`, but the second one is actually just comparing the two arguments.

Perhaps we could make this method public and test it directly?

Also worth noting that @MikeCollingwood is already going to do that (I hope) as part of https://github.com/govuk-one-login/ipv-core-back/pull/1367/files#r1394057532 - reusing this in the main CI scoring step.

Reply:
If you feel that the integer conversion is a distraction, you could add an extra method to the config service - `getIntegerSsmParameter` or similar, but not too fussed.

Reply:
I intentionally left the wrapper method in place to keep the `getFailExitCode()` more readable. Including the two calls to the config service in the predicate for the ternary got really busy. I was on the fence for a while though. Happy to ditch it if you think it's better.

Not too worried about the integer conversion as it's tucked away in the new class.

Will catch up with Mike about what he's up to and see if we can combine.

Reply:
@Joe-Edwards-GDS I've had a chat with Mike and he's happy to incorporate this into his work, as it's a larger piece.